mlir/lib/Dialect/EmitC/Transforms/TypeConversions.cpp

```diff
@@ -37,3 +37,28 @@ void mlir::populateEmitCSizeTTypeConversions(TypeConverter &converter) {
   converter.addTargetMaterialization(materializeAsUnrealizedCast);
   converter.addArgumentMaterialization(materializeAsUnrealizedCast);
 }
+
+/// Get an unsigned data type as wide as \p ty.
```

Comment:
Are `ptrdiff_t` and `size_t` guaranteed to be the same size?

Reply:
I would say it it is not guaranteed...

> The bit width of size_t is not less than 16. (since C99)

> The bit width of ptrdiff_t is not less than 17. (since C99) (until C23)
> The bit width of ptrdiff_t is not less than 16. (since C23)

Reply:
There is no spec on how wide these types have to be with respect to each other, see [here for C99](https://www.open-std.org/JTC1/SC22/WG14/www/docs/n1256.pdf) p. 266.

Reply:
So sould we reformulate the comment to something like `Get an unsigned data type for \p ty. The width of builtin integer types is guaranteed to be preserved.`?

Reply:
We don't need to mention the width, but the value must be preserved across casts, as they're mainly issued to adapt the signedness of a single MLIR `index`-type value in the C world. We can say the values will be preserved as long as they fit the types' ranges... but in this case we don't guarantee back and forth value-preserving casts for _any_ value that can be represented with these types. I should probably lower my expectation and be fine with this.